### PR TITLE
Redirected /oauth to /oauth/authorize

### PR DESCRIPTION
**Pull request by sukihub**
Having separate `/oauth` route seems redundant, since it only:
1. validates OAuth params 
2. performs login redirect

and both things are also done in `authorize` action.

I'd remove it completely, but that might break existing applications :)

Depends on #46

### CODE REVIEW

File: tests/bootstrap.php [hunk 1]
```diff
@@ -53,3 +54,6 @@
 Plugin::load('OAuth', [
     'path' => dirname(dirname(__FILE__)) . DS,
 ]);
+
+\Cake\Routing\DispatcherFactory::add('Routing');
+\Cake\Routing\DispatcherFactory::add('ControllerFactory');
```
Comment (dakota) [inline tests/bootstrap.php]: This might break applications using middleware

Comment (sukihub) [inline tests/bootstrap.php]: Actually, that's only bootstrap for tests, it still uses old dispatcher by default https://github.com/cakephp/cakephp/blob/master/src/TestSuite/IntegrationTestCase.php#L440 

I can try to update it, but it doesn't matter that much.

Comment (dakota) [inline tests/bootstrap.php]: Ok :)

**Comment by sukihub** [comment]
Since we've talked about making it backward incompatible, how do you feel about deleting oauth action? :)

**Comment by dakota** [comment]
@sukihub Kill it with fire :) It's not doing anything useful.

**Pull request merged**